### PR TITLE
Fix GCC's UBsan and update a list of disabled UBSan checks

### PR DESCRIPTION
**Pull request by ligurio**
Depends on https://github.com/tarantool/tarantool/issues/10699

### CODE REVIEW

File: cmake/compiler.cmake [hunk 1]
```diff
@@ -224,8 +224,11 @@ macro(enable_tnt_compile_flags)
     endif()
 
     if (ENABLE_UB_SANITIZER)
-        if (NOT CMAKE_COMPILER_IS_CLANG)
-            message(FATAL_ERROR "Undefined behaviour sanitizer only available for clang")
+        # UndefinedBehaviourSanitizer has been added to GCC since
+        # version 4.9.0, see https://gcc.gnu.org/gcc-4.9/changes.html.
+        if (CMAKE_COMPILER_IS_GNUCC AND
```
Comment (Buristan) [inline cmake/compiler.cmake:229]: Typo: s/if (/if(/

**Review by Buristan — APPROVED** [review]
Hi, Sergey!
Thanks for the patchset!
I'll proceed with the review per-patch below.

---

**[PATCH 1/4] cmake: replace UBsan flags whitelist by blacklist**
LGTM.

---

**[PATCH 2/4] cmake: enable UBsan checks**

LGTM.
Should it also mention #10228?

---

**[PATCH 3/4] cmake: enable UBSan check signed-integer-overflow**

LGTM.
Should it close #10228 or you have more subissues to resolve?

**[PATCH 4/4] cmake: fix build with GCC's UBSan**

LGTM, except a single nit below.

**Comment by ligurio** [comment]
>> [PATCH 2/4] cmake: enable UBsan checks

>> LGTM.
>> Should it also mention https://github.com/tarantool/tarantool/issues/10228?

Added #10228 to commit message

>> [PATCH 3/4] cmake: enable UBSan check signed-integer-overflow

>> LGTM.
>> Should it close https://github.com/tarantool/tarantool/issues/10228 or you have more subissues to resolve?

Added three issues and marked #10228 as to be closed.

**Review by grafin — APPROVED** [review]
Thanks for the patchset! LGTM.

**Review by Buristan — APPROVED** [review]

**Pull request merged**